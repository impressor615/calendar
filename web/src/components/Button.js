import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';


const Button = ({ children, color, onClick, active, ...rest }) => (
  <button
    type="button"
    onClick={onClick}
    {...rest}
    className={classnames(
      'btn',
      rest.className,
      { [color]: color },
      { active },
    )}>
    { children }
  </button>
);

Button.defaultProps = {
  color: '',
  active: false,
  onClick: () => {},
};

Button.propTypes = {
  color: PropTypes.oneOf(['', 'primary', 'secondary']),
  onClick: PropTypes.func,
  children: PropTypes.node.isRequired,
  active: PropTypes.bool,
}

export default Button;
