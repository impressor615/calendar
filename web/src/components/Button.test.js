import React from 'react';

import Button from 'components/Button';


describe("<Button />", () => {
  const onClick = jest.fn();
  const props = {
    children: 'button',
    active: false,
    onClick,
  };

  it('should be rendered properly based on its props', () => {
    const wrapper = shallow(<Button {...props} />);
    expect(wrapper.find('.btn')).toHaveLength(1);
    expect(wrapper.find('.btn').text()).toEqual(props.children);

    wrapper.setProps({ active: true });
    expect(wrapper.find('.btn.active')).toHaveLength(1);

    wrapper.setProps({ color: 'primary' });
    expect(wrapper.find('.btn.primary')).toHaveLength(1);

    wrapper.setProps({ color: 'secondary' });
    expect(wrapper.find('.btn.secondary')).toHaveLength(1);
  });

  it('onClick should be called when clicking the button', () => {
    const wrapper = shallow(<Button {...props} />);
    wrapper.find('.btn').simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should be matched with the snapshot', () => {
    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
