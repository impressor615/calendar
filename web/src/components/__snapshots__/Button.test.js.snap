// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be matched with the snapshot 1`] = `
<button
  className="btn"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  button
</button>
`;
